src/parser/ast/loops.rs

```diff
@@ -83,37 +80,45 @@ pub enum ForBody {
 
 #[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize, JsonSchema)]
 #[serde(rename_all = "snake_case")]
-pub struct DoWhileLoop {
+pub struct DoWhileStatement {
     pub r#do: Span,                                                // `do`
-    pub body: DoWhileBody,                                         // `{ ... }`
+    pub body: Box<Statement>,                                      // `{ ... }`
     pub r#while: Span,                                             // `while`
     pub condition: SemicolonTerminated<Parenthesized<Expression>>, // `( *expression* )`
 }
 
-#[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize, JsonSchema)]
-#[serde(rename_all = "snake_case", tag = "type", content = "value")]
-pub enum DoWhileBody {
-    Statement(Box<Statement>),
-    Braced(Braced<Vec<Statement>>),
-}
-
 #[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize, JsonSchema)]
 #[serde(rename_all = "snake_case")]
-pub struct WhileLoop {
+pub struct WhileStatement {
     pub r#while: Span,                        // `while`
     pub condition: Parenthesized<Expression>, // `( *expression* )`
-    pub body: WhileBody,                      // `{ ... }`
+    pub body: WhileStatementBody,             // `{ ... }`
 }
 
 #[derive(Debug, PartialEq, Eq, Clone, Deserialize, Serialize, JsonSchema)]
 #[serde(rename_all = "snake_case", tag = "type", content = "value")]
-pub enum WhileBody {
+pub enum WhileStatementBody {
     Statement(Box<Statement>),
-    Braced(Braced<Vec<Statement>>),
     Block {
```

Comment:
as discussed privately, a braced vec of statements is a statement on it self, so i doesn't make sense to have it here.

and mixing braced-vec-of-statements and single-statement is allowed, so there's no need to have a special node for it.